### PR TITLE
[cmake] Don't append target sysroot in native toolchainfile

### PR DESCRIPTION
**Pull request by sundermann**
## Description

Target sysroot libraries are ending up inside the native build causing warnings and apparently errors. (#24552)

## How has this been tested?

webOS only

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: tools/depends/native/Toolchain-Native.cmake.in [hunk 1]
```diff
@@ -51,7 +51,7 @@ set(CMAKE_EXE_LINKER_FLAGS "@host_includes@ -L@prefix@/@tool_dir@/lib")
 set(CMAKE_FIND_ROOT_PATH @prefix@/@tool_dir@)
 set(CMAKE_LIBRARY_PATH @prefix@/@tool_dir@/lib)
 if(NOT "@use_toolchain@" STREQUAL "")
-  list(APPEND CMAKE_FIND_ROOT_PATH @use_toolchain@ @use_toolchain@/@use_host@ @use_toolchain@/@use_host@/sysroot @use_toolchain@/@use_host@/sysroot/usr @use_toolchain@/@use_host@/libc @use_toolchain@/lib/@use_host@/sysroot @use_toolchain@/usr @use_toolchain@/sysroot/usr)
+  list(APPEND CMAKE_FIND_ROOT_PATH @use_toolchain@ @use_toolchain@/lib/@use_host@/sysroot @use_toolchain@/usr @use_toolchain@/sysroot/usr)
```
Comment (fuzzard) [inline tools/depends/native/Toolchain-Native.cmake.in:54]: Should anything that use @use_host@ be removed entirely? eg @use_toolchain@/lib/@use_host@/sysroot as well?

**Comment by sundermann** [comment]
@fuzzard only removed what is causing trouble right now. But IMO only `list(APPEND CMAKE_FIND_ROOT_PATH @use_toolchain@)` would be enough. Not sure about other platforms though. I guess at least Android uses this code path too

File: tools/depends/native/Toolchain-Native.cmake.in (hunk 1 above)
Comment (sundermann) [reply tools/depends/native/Toolchain-Native.cmake.in:54]: Well I think so, but these folders don't exist in the webOS toolchain so they don't harm

**Comment by fuzzard** [comment]
from an apple perspective, @use_toolchain@ is enough. We only use that, and then we explicitly use @host_sysroot@ after for everything required.

Looks like android might need @use_toolchain@/sysroot/usr, at the least its a valid path

File: tools/depends/native/Toolchain-Native.cmake.in (hunk 1 above)
Comment (sundermann) [reply tools/depends/native/Toolchain-Native.cmake.in:54]: Let's see first if this breaks any other platform but I think also the line below is not needed as `CMAKE_FIND_ROOT_PATH` should already account for it.

**Comment by sundermann** [comment]
Can you check the architecture of the files inside `@use_toolchain@/sysroot/usr` on Android?

**Comment by fuzzard** [comment]
As i look at it further, looks like its target related too.

```
brent@Brents-MacBook-Pro-2 depends % ls /Users/brent/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/lib 
aarch64-linux-android	arm-linux-androideabi	i686-linux-android	x86_64-linux-android
```

**Comment by sundermann** [comment]
Run `file` on some random library inside one of these folders just to be sure

**Comment by fuzzard** [comment]
```
brent@Brents-MacBook-Pro-2 depends % file /Users/brent/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/lib/aarch64-linux-android/33/libGLESv2.so
/Users/brent/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/lib/aarch64-linux-android/33/libGLESv2.so: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), static-pie linked, BuildID[md5/uuid]=9c7fc892595fbf34bcc083c2e2dc6299, with debug_info, not stripped
brent@Brents-MacBook-Pro-2 depends % file /Users/brent/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/lib/x86_64-linux-android/33/libGLESv2.so 
/Users/brent/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/lib/x86_64-linux-android/33/libGLESv2.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), static-pie linked, BuildID[md5/uuid]=d78be2a702632e81347d2fb4856a907a, with debug_info, not stripped
```

Arch specific

**Comment by sundermann** [comment]
Well yeah then that should not be inside the native file

**Comment by sundermann** [comment]
So let's see what happens if we remove everything but `@use_toolchain@`

**Review by fuzzard — APPROVED** [review]
CI looks good, let's merge

**Pull request merged**